### PR TITLE
Convert authentication session entities into interface

### LINKED ISSUE #9738: Convert authentication session entities into interface
### Description

Convert implementation of Map*Entity into interfaces in authentication session area

### PR DESCRIPTION
**Pull request by martin-kanis**
Closes #9738.

### CODE REVIEW

**Review by mhajas — CHANGES_REQUESTED** [review]
Thank for the PR @martin-kanis, I added two comments into code

**Review by mhajas — APPROVED** [review]
Thanks @martin-kanis

File: model/map/src/main/java/org/keycloak/models/map/authSession/MapAuthenticationSessionEntity.java [hunk 1]
```diff
@@ -16,129 +16,77 @@
...
-
-    private Map<String, String> clientNotes= new ConcurrentHashMap<>();;
-    private Map<String, String> authNotes = new ConcurrentHashMap<>();;
-    private Set<String> requiredActions = new HashSet<>();
-    private Map<String, String> userSessionNotes = new ConcurrentHashMap<>();
-
-    public Map<String, String> getUserSessionNotes() {
-        return userSessionNotes;
+        @Override
+        public void addExecutionStatus(String authenticator, AuthenticationSessionModel.ExecutionStatus status) {
```
[the remark this answers is not in the document]
Comment (martin-kanis) [reply model/map/src/main/java/org/keycloak/models/map/authSession/MapAuthenticationSessionEntity.java]: Thank you for the suggestion. I included it.

[the remark this answers is not in the document]
Comment (mhajas) [reply model/map/src/main/java/org/keycloak/models/map/authSession/MapAuthenticationSessionEntity.java]: Done

File: model/map/src/main/java/org/keycloak/models/map/authSession/MapRootAuthenticationSessionEntity.java [hunk 2]
```diff
@@ -16,89 +16,71 @@
...
+
+        @Override
+        public void setId(String id) {
+            if (this.id != null) throw new IllegalStateException("Id cannot be changed");
+            this.id = id;
+            this.updated |= id != null;
+        }
+
+        @Override
+        public MapAuthenticationSessionEntity removeAuthenticationSession(String tabId) {
```
[the remark this answers is not in the document]
Comment (mhajas) [reply model/map/src/main/java/org/keycloak/models/map/authSession/MapRootAuthenticationSessionEntity.java]: Done

[the remark this answers is not in the document]
Comment (martin-kanis) [reply model/map/src/main/java/org/keycloak/models/map/authSession/MapRootAuthenticationSessionEntity.java]: Changed.

**Review by hmlnarik — APPROVED** [review]
Approving per @mhajas 's approval

**Pull request merged**